src/Avalonia.Controls.ColorPicker/Themes/Fluent/ColorPicker.xaml

```diff
@@ -0,0 +1,82 @@
+﻿<Styles xmlns="https://github.com/avaloniaui"
+        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+        xmlns:local="using:Avalonia.Controls"
+        xmlns:converters="using:Avalonia.Controls.Converters"
+        x:CompileBindings="False">
+
+  <Styles.Resources>
+    <converters:ToBrushConverter x:Key="ToBrush" />
+    <converters:CornerRadiusFilterConverter x:Key="LeftCornerRadiusFilterConverter" Filter="TopLeft, BottomLeft"/>
+  </Styles.Resources>
+
+  <Style Selector="ColorPicker">
+    <Setter Property="CornerRadius" Value="{DynamicResource ControlCornerRadius}" />
+    <Setter Property="Height" Value="32" />
+    <Setter Property="Width" Value="64" />
+    <Setter Property="MinWidth" Value="64" />
+    <Setter Property="Template">
+      <ControlTemplate>
+        <DropDownButton CornerRadius="{TemplateBinding CornerRadius}"
+                        Height="{TemplateBinding Height}"
+                        Width="{TemplateBinding Width}"
+                        HorizontalContentAlignment="Stretch"
+                        VerticalContentAlignment="Stretch"
+                        Padding="0,0,10,0"
+                        UseLayoutRounding="False">
+          <DropDownButton.Styles>
+            <Style Selector="FlyoutPresenter.NoPadding">
+              <Setter Property="Padding" Value="0" />
+            </Style>
+          </DropDownButton.Styles>
+          <DropDownButton.Content>
+            <!-- Preview color -->
+            <Panel>
+              <Border Background="{StaticResource CheckeredBackgroundBrush}"
+                      CornerRadius="{TemplateBinding CornerRadius, Converter={StaticResource LeftCornerRadiusFilterConverter}}"
+                      HorizontalAlignment="Stretch"
+                      VerticalAlignment="Stretch"
+                      Margin="1,1,0,1" />
+              <Border Background="{Binding HsvColor, ElementName=FlyoutColorView, Converter={StaticResource ToBrush}, Mode=OneWay}"
+                      CornerRadius="{TemplateBinding CornerRadius, Converter={StaticResource LeftCornerRadiusFilterConverter}}"
+                      HorizontalAlignment="Stretch"
+                      VerticalAlignment="Stretch"
+                      Margin="1,1,0,1" />
+            </Panel>
+          </DropDownButton.Content>
+          <DropDownButton.Flyout>
+            <Flyout FlyoutPresenterClasses="NoPadding">
+              <!-- Skip the following:
+                - CustomPaletteColors
+              -->
+              <ColorView x:Name="FlyoutColorView"
+                         Color="{Binding $parent[ColorPicker].Color, Mode=TwoWay}"
```

Comment:
@maxkatz6 Binding here with `Color` and `HsvColor` below works a little, but has a lot of issues. When selecting a color in the ColorView the value jumps around and is incorrect half the time. I need ideas for how to fix this, work around it, or if this is really an issue in the binding system. I'm not sure where to go from here.

Also note that binding to the TemplatedParent doesn't seem to work either which is why I went with the `$parent[ColorPicker].Color` syntax.

Reply:
I did not have a deep look, but I see a possible problem:

0. Previous **slider** HSV value is ```hsva(48, 0.6701030927835051, 0.3803921568627451, 1)``` and RGB - ```#ff615420```.
1. User [changes HSV property](https://github.com/AvaloniaUI/Avalonia/blob/bd8f7446d5136216919e0a59468fc1f0e83bd585/src/Avalonia.Controls.ColorPicker/ColorSlider/ColorSlider.cs#L412) of the **slider**. New value is hsva(48, 0.6701030927835051, 0.35163934426229515, 1). This value is sent with a binding notification to the **view**.
2. **View** [receives HSV and changes RGB](https://github.com/AvaloniaUI/Avalonia/blob/bd8f7446d5136216919e0a59468fc1f0e83bd585/src/Avalonia.Controls.ColorPicker/ColorView/ColorView.cs#L117) to ```#ff5a4e1e``` and sends binding notification to the **picker**. **disableUpdates** is true on the **view**, but false on the **picker**.
3. **Picker** [receives RGB and changes HSV](https://github.com/AvaloniaUI/Avalonia/blob/bd8f7446d5136216919e0a59468fc1f0e83bd585/src/Avalonia.Controls.ColorPicker/ColorView/ColorView.cs#L104) to ```hsva(48.00000000000001, 0.6666666666666666, 0.3529411764705882, 1)```. **disableUpdates** is true on the **view**, and true on the **picker**.
4. At this point ColorPicker.Hsv propagates through bindings code, hard to tell what's going on from the stacktrace.
5. Finally **Spectrum** receives HSV value, that was calculated on step number 3 inside of ColorPicker. Not the original one. And spectrum converts it to the RGB once more.

Reply:
As you can see, color from the slider goes quite a long way. And gets recalculated couple of times on the way.
**disableUpdates** field is useless when value goes from View to the Picker.

Reply:
Probably it would be easier to add some syncing code inside of the ColorPicker for (Hsv)Color and CustomPaletteColors props. As I don't see any obvious problem in bindings that can be fixed.

Reply:
Yes, this definitely looks like a circular property changed issue as you describe. I thought I designed around it but missed something.

For 0-3, I agree, that is how it should work. Note that ONLY the ColorView is responsible for the synchronization with RBG/HSV and Color/HsvColor properties. This is because all primitives are bound only through the HsvColor property. If a primitive needs RGB color it calculates it itself internally and separately. This means we should never have to worry about both Color and HsvColor binding changes occurring in a bad order or repeatedly updating each other. This is known to work just fine because the ColorView functions perfectly. So we know the issue is with ColorPicker.

Now, here is my original thinking that I realize is now incorrect: (thanks for coaxing in the right direction)

In the ColorPicker, the Color and HsvColor are bound directly to the ColorView. So if the ColorView is correct -- and it is -- the ColorPicker should just automatically get the values. Note that the ColorPicker has zero code-behind so I was stupidly thinking it shouldn't be propagating any changes back to the ColorView or doing RGB<->HSV sync. Even if it was, the ColorView shouldn't do anything because the values would be equal to the one that triggered the binding update to begin with. This was likely my mistake in hindsight.

Of course ColorPicker is derived from ColorView... so they both have the color sync logic running simultaneously. That was untested and was never intended. This seems to be the core problem. I'll have to add a way to disable the RGB<->HSV color sync logic internally in derived controls like ColorPicker or make this logic behave correctly with derived controls.

Anyway, thanks for pointing me in the right direction (calling out disableUpdates being different made me realize). I'll debug this later with lots of sleep.

Reply:
It seems that wasn't it. I also changed to normal bindings and they don't work at all. I think standard bindings can't reach across the Flyout/Popup to the root control. Even More, the control becomes unresponsive after the flyout closes.

Reply:
@maxkatz6 I confirmed two things:

1. The 'freezing' of the control that was happening when the flyout closes is now fixed with latest master. Not sure what that was
2. I rebuilt the template and put the ColorView outside of the Flyout just to test the bindings -- it worked. This proves the issue is that TemplateBinding in Avalonia does NOT work across a Popup/Flyout. I'll need help to fix this.

Reply:
#8367 was fixed resolving this conversation